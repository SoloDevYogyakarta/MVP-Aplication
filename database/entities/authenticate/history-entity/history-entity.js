"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.userHistoryEntity = exports.UserHistoryEntity = void 0;
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class UserHistoryEntity extends base_entity_1.BaseEntity {
}
exports.UserHistoryEntity = UserHistoryEntity;
exports.userHistoryEntity = entity_1.sequelize.define('HISTORY', {
    plat_number: sequelize_1.DataTypes.STRING,
    phone_number: sequelize_1.DataTypes.STRING,
    name: sequelize_1.DataTypes.STRING,
    address: sequelize_1.DataTypes.STRING,
    user_id: sequelize_1.DataTypes.INTEGER,
}, {
    tableName: 'HISTORY',
    schema: 'AUTHENTICATE',
    timestamps: false,
});
//# sourceMappingURL=history-entity.js.map