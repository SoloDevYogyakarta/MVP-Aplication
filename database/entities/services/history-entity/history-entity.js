"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.historyEntity = exports.HistoryEntity = void 0;
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class HistoryEntity extends base_entity_1.BaseEntity {
}
exports.HistoryEntity = HistoryEntity;
exports.historyEntity = entity_1.sequelize.define('HISTORY', {
    name: sequelize_1.DataTypes.STRING,
    title: sequelize_1.DataTypes.STRING,
    desc: sequelize_1.DataTypes.STRING,
    price: sequelize_1.DataTypes.INTEGER,
    order_id: sequelize_1.DataTypes.INTEGER,
}, {
    tableName: 'HISTORY',
    schema: 'SERVICES',
});
//# sourceMappingURL=history-entity.js.map