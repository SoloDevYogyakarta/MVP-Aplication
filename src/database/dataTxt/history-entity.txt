{"id":11,"public_id":"MLf4_sOkio9GAfkIwj4tk","name":"Fantastic Steel Bacon","desc":"New range of formal shirts are designed keeping you in mind. With fits and styling that will make you stand apart","user_id":"Hk3YcUFZgDO1e9H3u9B16","updatedAt":"2023-10-26T06:54:52.587Z","createdAt":"2023-10-26T06:54:52.587Z"}