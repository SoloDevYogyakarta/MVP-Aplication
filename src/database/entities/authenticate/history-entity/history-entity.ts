import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class UserHistoryEntity extends BaseEntity {
  plat_number: string;
  phone_number: string;
  name: string;
  address: string;
  user_id: number;
}

export type UserHistoryCreationAttribute = Optional<UserHistoryEntity, 'id'>;
export interface UserHistoryInstance
  extends Model<UserHistoryCreationAttribute, UserHistoryEntity>,
    UserHistoryEntity {}
export const userHistoryEntity = sequelize.define<UserHistoryInstance>(
  'HISTORY',
  {
    plat_number: DataTypes.STRING,
    phone_number: DataTypes.STRING,
    name: DataTypes.STRING,
    address: DataTypes.STRING,
    user_id: DataTypes.INTEGER,
  } as ModelAttributes<UserHistoryInstance>,
  {
    tableName: 'HISTORY',
    schema: 'AUTHENTICATE',
    timestamps: false,
  },
);
