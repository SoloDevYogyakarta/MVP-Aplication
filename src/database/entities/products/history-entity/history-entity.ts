import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class HistoryEntity extends BaseEntity {
  type: string;
  date: Date;
  product_id: string;
  user_id: string;
}

export type HistoryCreationAttribute = Optional<HistoryEntity, 'id'>;
export interface HistoryInstance
  extends Model<HistoryCreationAttribute, HistoryEntity>,
    HistoryEntity {}
export const historyEntity = sequelize.define<HistoryInstance>(
  'HISTORY',
  {
    public_id: DataTypes.STRING,
    type: DataTypes.STRING,
    date: DataTypes.DATE,
    product_id: DataTypes.STRING,
    user_id: DataTypes.STRING,
  } as ModelAttributes<HistoryInstance>,
  {
    tableName: 'HISTORY',
    schema: 'PRODUCTS',
    timestamps: false,
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
