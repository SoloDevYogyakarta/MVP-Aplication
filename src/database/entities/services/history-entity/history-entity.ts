import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class HistoryEntity extends BaseEntity {
  name: string;
  desc: string;
  user_id: string;
}

export type HistoryCreationAttribute = Optional<HistoryEntity, 'id'>;
export interface HistoryInstance
  extends Model<HistoryCreationAttribute, HistoryEntity>,
    HistoryEntity {}
export const historyEntity = sequelize.define<HistoryInstance>(
  'HISTORY',
  {
    public_id: DataTypes.STRING,
    name: DataTypes.STRING,
    desc: DataTypes.STRING,
    user_id: DataTypes.STRING,
  } as ModelAttributes<HistoryInstance>,
  {
    tableName: 'HISTORY',
    schema: 'SERVICES',
  },
);
