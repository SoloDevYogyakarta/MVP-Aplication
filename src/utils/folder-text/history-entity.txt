{"id":16,"name":"Handmade Wooden Mouse","title":"Bronze","desc":"The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design","price":69653,"order_id":8,"updatedAt":"2023-11-07T07:15:16.412Z","createdAt":"2023-11-07T07:15:16.412Z"}