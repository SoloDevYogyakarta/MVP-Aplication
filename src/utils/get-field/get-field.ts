import { GetFieldType } from 'src/types/get-field.type';
import { readpath } from '../system/system';

export function getField(filename?: GetFieldType) {
  try {
    if (filename === 'fake') {
      return readpath(`../../database/dataTxt/${filename}.txt`);
    }
    return JSON.parse(readpath(`../../database/dataTxt/${filename}.txt`));
  } catch (err) {
    throw new Error(err);
  }
}
