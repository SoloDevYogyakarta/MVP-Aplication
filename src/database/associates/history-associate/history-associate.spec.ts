import { ModelCtor } from 'sequelize';
import { HistoryInstance } from '../../../database/entities/services/history-entity/history-entity';
import { getField } from '../../../utils/get-field/get-field';
import {
  historyAssociate,
  historyAttribute,
  historyInclude,
} from './history-associate';

describe('HistoryAssociate', () => {
  let public_id!: string;
  let entity: ModelCtor<HistoryInstance>;

  beforeEach(() => {
    entity = historyAssociate;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  try {
    public_id = getField('history-entity').public_id;
  } catch (err) {
    // empty
  }

  if (public_id) {
    it('findOne', async () => {
      const findOne = await entity.findOne({
        where: { public_id },
        attributes: historyAttribute,
        include: historyInclude,
      });
      expect(findOne.public_id).toEqual(public_id);
    });

    it('findAll', async () => {
      const findAll = await entity.findAll({
        attributes: historyAttribute,
        include: historyInclude,
      });
      expect(findAll.length).not.toEqual(0);
    });
  }
});
