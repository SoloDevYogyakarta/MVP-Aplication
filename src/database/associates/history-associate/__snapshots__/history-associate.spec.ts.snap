// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryAssociate render correctly 1`] = `[Function]`;
